pkg/noderesourcetopology/filter_test.go

```diff
@@ -577,6 +586,136 @@ func TestNodeResourceTopologyMultiContainerPodScope(t *testing.T) {
 	}
 }
 
+func TestNodeResourceTopologyMultiContainerContainerScope(t *testing.T) {
+	nodeTopologies := []*topologyv1alpha1.NodeResourceTopology{
+		{
+			ObjectMeta:       metav1.ObjectMeta{Name: "host0"},
+			TopologyPolicies: []string{string(topologyv1alpha1.SingleNUMANodeContainerLevel)},
+			Zones: topologyv1alpha1.ZoneList{
+				{
+					Name: "node-0",
+					Type: "Node",
+					Resources: topologyv1alpha1.ResourceInfoList{
+						MakeTopologyResInfo(cpu, "32", "30"),
+						MakeTopologyResInfo(memory, "64Gi", "60Gi"),
+						MakeTopologyResInfo(hugepages2Mi, "384Mi", "384Mi"),
+						MakeTopologyResInfo(nicResourceName, "16", "16"),
+					},
+				},
+				{
+					Name: "node-1",
+					Type: "Node",
+					Resources: topologyv1alpha1.ResourceInfoList{
+						MakeTopologyResInfo(cpu, "32", "32"),
+						MakeTopologyResInfo(memory, "64Gi", "64Gi"),
+						MakeTopologyResInfo(hugepages2Mi, "512Mi", "512Mi"),
+						MakeTopologyResInfo(nicResourceName, "32", "32"),
+					},
+				},
+			},
+		},
+	}
+
+	nodes := make([]*v1.Node, len(nodeTopologies))
+	for i := range nodes {
+		nodes[i] = makeNodeFromNodeResourceTopology(nodeTopologies[i])
+	}
+
+	tests := []struct {
+		name       string
+		pod        *v1.Pod
+		node       *v1.Node
+		nrts       []*topologyv1alpha1.NodeResourceTopology
+		avail      []resourceDescriptor
+		wantStatus *framework.Status
+	}{
+		{
+			name: "gu pod, overallocation",
+			pod: makeMultiContainerPodWithResourceList("testpod1",
+				[]v1.ResourceList{
+					{
+						v1.ResourceCPU:                   resource.MustParse("24"),
+						v1.ResourceMemory:                resource.MustParse("48Gi"),
+						v1.ResourceName(hugepages2Mi):    resource.MustParse("384Mi"),
+						v1.ResourceName(nicResourceName): resource.MustParse("24"),
+					},
+					{
+						v1.ResourceCPU:                   resource.MustParse("24"),
+						v1.ResourceMemory:                resource.MustParse("48Gi"),
+						v1.ResourceName(hugepages2Mi):    resource.MustParse("384Mi"),
+						v1.ResourceName(nicResourceName): resource.MustParse("24"),
+					},
+					{
+						v1.ResourceCPU:                   resource.MustParse("24"),
+						v1.ResourceMemory:                resource.MustParse("48Gi"),
+						v1.ResourceName(hugepages2Mi):    resource.MustParse("384Mi"),
+						v1.ResourceName(nicResourceName): resource.MustParse("24"),
+					},
+				},
+			),
+			node: nodes[0],
+			nrts: []*topologyv1alpha1.NodeResourceTopology{
+				nodeTopologies[0],
+			},
+			avail:      []resourceDescriptor{},
+			wantStatus: framework.NewStatus(framework.Unschedulable, "cannot align pod: testpod1"),
```

Comment:
The status message here is incorrect.
The ```Filter``` function will check this pod against the containerScope and not podScope, because the ```nodeTopologies[0]``` (line 658) object has ```topologyv1alpha1.SingleNUMANodeContainerLevel``` hence the function will return ```"cannot align container..."```